### PR TITLE
feat: Use different default extensions for Prettier depending on its version

### PR DESCRIPTION
**Pull request by okonet**
The official Prettier documentation suggest using Prettiers for CSS, JSON and Markdown files: https://prettier.io/docs/en/precommit.html#option-1-lint-staged-https-githubcom-okonet-lint-staged

Since language support was introduced gradually, this PR implements the different default extensions based on the the Prettier version from `package.json`.

### CODE REVIEW

**Comment by okonet** [comment]
@sapegin one thing: I could not get this working when running locally in the `mrm-tasks` reposotory. Was I doing it wrong? Any way to test it locally since tests are passing...

**Comment by okonet** [comment]
Wow lots of tests failed... @sapegin can you please check what's going on?

File: packages/mrm-task-lintstaged/index.js [hunk 1]
```diff
@@ -32,9 +33,21 @@ function task(config) {
 
 		// Prettier
 		if (pkg.get('devDependencies.prettier') && !pkg.get('devDependencies.eslint-plugin-prettier')) {
+			const prettierVersion = semver.coerce(pkg.get('devDependencies.prettier'));
 			const script = pkg.getScript('format');
 			const exts = getExtsFromCommand(script);
-			newRules.push([extsToGlob(exts, prettierExtensions), 'prettier --write']);
+			let suggestedExtensions = prettierExtensions;
```
Comment (sapegin) [inline packages/mrm-task-lintstaged/index.js]: We shouldn't overwrite user config value.

File: packages/mrm-task-lintstaged/index.js [hunk 2]
```diff
@@ -32,9 +33,21 @@ function task(config) {
...
+			if (semver.satisfies(prettierVersion, '1.4.0 - 1.5.0')) {
+				// CSS was added in 1.4
+				suggestedExtensions = '.js,css';
```
Comment (sapegin) [inline packages/mrm-task-lintstaged/index.js]: Shouldn't it be `.js,.css`?

**Comment by sapegin** [comment]
I guess I need to update mrm-core in master ;-/

**Review by sapegin — CHANGES_REQUESTED** [review]

File: packages/mrm-task-lintstaged/index.js (hunk 2 above)
Comment (okonet) [reply packages/mrm-task-lintstaged/index.js]: Not according to tests. It’s a weird format anyways. I was expecting it to be a glob or an array.

File: packages/mrm-task-lintstaged/index.js (hunk 1 above)
Comment (okonet) [reply packages/mrm-task-lintstaged/index.js]: I thought `exts` is user extensions. And the second argument is default value. Tests also passing for user extensions.

**Comment by sapegin** [comment]
Try to merge master, tests should be fine.

File: packages/mrm-task-lintstaged/index.js (hunk 1 above)
Comment (sapegin) [reply packages/mrm-task-lintstaged/index.js]: `exsts` are extensions defined in the `package.json` and `prettierExtensions` is from a config which has a default value.

**Comment by okonet** [comment]
@sapegin added a fix and a test. Please review.

File: packages/mrm-task-lintstaged/index.js [hunk 3]
```diff
@@ -34,7 +40,23 @@ function task(config) {
 		if (pkg.get('devDependencies.prettier') && !pkg.get('devDependencies.eslint-plugin-prettier')) {
 			const script = pkg.getScript('format');
 			const exts = getExtsFromCommand(script);
-			newRules.push([extsToGlob(exts, prettierExtensions), 'prettier --write']);
+			let suggestedExtensions = prettierExtensions;
+			// Use different extensions for Prettier depdnding on its version
+			// but only do so if use didn't override default extensions
```
Comment (sapegin) [inline packages/mrm-task-lintstaged/index.js]: use → user

**Review by sapegin — CHANGES_REQUESTED** [review]
Just a typo in a comment ;-)

File: packages/mrm-task-lintstaged/index.js (hunk 3 above)
Comment (okonet) [reply packages/mrm-task-lintstaged/index.js]: There are actually 2 typos there 🤦‍♂️

Comment (okonet) [reply packages/mrm-task-lintstaged/index.js]: `depdnding` 🤦‍♂️

Comment (okonet) [reply packages/mrm-task-lintstaged/index.js]: Fixed. Please review.

**Review by sapegin — APPROVED** [review]

**Comment by sapegin** [comment]
Thanks, merged!

**Pull request merged**